File: backend/app/schemas/hotel.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class HotelCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    city: str = Field(min_length=1, max_length=100)
    address: str = Field(min_length=1, max_length=500)
    description: str | None = None
    stars: int = Field(ge=1, le=5)
    latitude: Decimal = Field(ge=Decimal("-90"), le=Decimal("90"))
    longitude: Decimal = Field(ge=Decimal("-180"), le=Decimal("180"))


class HotelUpdate(HotelCreate):
    pass


class HotelListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    city: str
    address: str
    description: str | None
    stars: int
    latitude: Decimal
    longitude: Decimal
    created_at: datetime
    avg_rating: float | None
    reviews_count: int
    min_price: Decimal | None
    cover_image: None = None
    is_favorite: bool | None


class HotelImage(BaseModel):
    id: int
    url: str
    sort_order: int


class HotelDetail(HotelListItem):
    images: list[HotelImage]


class HotelPage(BaseModel):
    items: list[HotelListItem]
    total: int
    page: int
    size: int

File: frontend/src/types/hotel.ts
export type HotelSort = 'created_at' | 'stars' | 'avg_rating'

export interface HotelListItem {
  id: number
  name: string
  city: string
  address: string
  description: string | null
  stars: number
  latitude: string
  longitude: string
  created_at: string
  avg_rating: number | null
  reviews_count: number
  min_price: string | null
  cover_image: string | null
  is_favorite: boolean | null
}

export interface HotelImage {
  id: number
  url: string
  sort_order: number
}

export interface HotelDetail extends HotelListItem {
  images: HotelImage[]
}

export interface HotelPage {
  items: HotelListItem[]
  total: number
  page: number
  size: number
}

export interface HotelListParams {
  city?: string
  stars?: number
  sort?: HotelSort
  order?: 'asc' | 'desc'
  page?: number
  size?: number
}

export interface HotelWritePayload {
  name: string
  city: string
  address: string
  description?: string | null
  stars: number
  latitude: number
  longitude: number
}
